### ISSUE TITLE
`no-unexpected-multiline` flags opening parentheses sent as first argument to a function call

### ISSUE THREAD

**Issue by dtinth** [eslint/eslint#4658]
I am using `eslint@1.10.3`.

When sending a parenthesized expression on its own line to a function to a function call, `no-unexpected-multiline` rule flags that expression.

```js
// example.js
module.exports = compose(
  (process.env.NODE_ENV === 'production'
    ? generateProductionConfig
    : generateDevelopmentConfig
  ),
  generateBaseConfig
)()
```

It is expected that the above code does not produce a warning, as it is clear that we are inside a call to function `compose`. There can be no ambiguity here.

However, the `no-unexpected-multiline` rule flags the newline between two `(`s as unexpected.

```
$ eslint example.js --rule 'no-unexpected-multiline: 2'

example.js
  3:3  error  Unexpected newline between function and ( of function call  no-unexpected-multiline

✖ 1 problem (1 error, 0 warnings)
```

When sending as a subsequent parameter, however, there is no warning (this is correct behavior).

```js
// example.js
module.exports = pipe(
  generateBaseConfig,
  (process.env.NODE_ENV === 'production'
    ? generateProductionConfig
    : generateDevelopmentConfig
  )
)()
```

```
$ eslint example.js --rule 'no-unexpected-multiline: 2'
# no warnings generated
```

Inside an array there is also no warning (this is correct behavior).

```js
// example.js
module.exports = compose([
  (process.env.NODE_ENV === 'production'
    ? generateProductionConfig
    : generateDevelopmentConfig
  ),
  generateBaseConfig,
])()
```

```
$ eslint example.js --rule 'no-unexpected-multiline: 2'
# no warnings generated
```

## Minimal Test Case

Should not produce a warning:

```js
f(
  (x)
)
```

**Comment by nzakas** [comment]
Looks like we are confusing the open paren as part of the function call.
